network/src/connectivity_manager/mod.rs

```diff
@@ -269,15 +280,20 @@ where
 
         loop {
             self.event_id = self.event_id.wrapping_add(1);
-            ::futures::select! {
-                _ = self.ticker.select_next_some() => {
+            futures::select! {
+                _ = ticker.select_next_some() => {
                     self.check_connectivity(&mut pending_dials).await;
                 },
                 req = self.requests_rx.select_next_some() => {
                     self.handle_request(req);
                 },
-                notif = self.connection_notifs_rx.select_next_some() => {
-                    self.handle_control_notification(notif);
+                maybe_notif = self.connection_notifs_rx.next() => {
+                    // Shutdown the connectivity manager when the PeerManager
+                    // shuts down.
+                    match maybe_notif {
+                        Some(notif) => self.handle_control_notification(notif),
+                        None => break,
```

Comment:
Btw, what happens to the pending dials and all at this point?  I guess they're really in the TransportHandler, so they should be shut down at this point

Reply:
yeah, if we're going to start spinning up and down network instances dynamically, we'd probably have to rethink the graceful shutdown process. for now, this is mostly for tests, since the peer manager will never shutdown in prod